### PR TITLE
CRS-131 Implement Example 20 test

### PR DESCRIPTION
**Pull request by joelstobart-moj**
- Implement the Example 20 tests
- Replace aggregate with combine
- Append all

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/OffenderSentenceProfileCalculationService.kt [hunk 1]
```diff
@@ -44,20 +52,23 @@ class OffenderSentenceProfileCalculationService(
...
 
-  fun aggregate(offenderSentenceProfile: OffenderSentenceProfile): OffenderSentenceProfile {
+  fun combine(offenderSentenceProfile: OffenderSentenceProfile): OffenderSentenceProfile {
     return when (offenderSentenceProfile.sentences.size) {
       0 -> throw NoSentencesProvidedException("At least one sentence must be provided")
       1 -> offenderSentenceProfile
       else -> {
-        aggregateMultiple(offenderSentenceProfile)
+        val workingOffenderSentenceProfile = combinationService.combineConsecutiveSentences(offenderSentenceProfile)
+        for (sentence in workingOffenderSentenceProfile.sentences) {
```
Comment (yasinmustafa) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/OffenderSentenceProfileCalculationService.kt:61]: could use a more functional way of looping rather the for loops

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SentenceCombinationService.kt [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    }
+  }
+
+  private fun associateExistingLinksToNewSentence(
+    originalSentence: Sentence,
+    replacementSentence: Sentence,
+    offenderSentenceProfile: OffenderSentenceProfile
+  ) {
+
+    for (sentence in offenderSentenceProfile.sentences) {
```
Comment (yasinmustafa) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SentenceCombinationService.kt:111]: same - could use a more functional way of looping rather the for loops. Happy  to defer

**Review by yasinmustafa — APPROVED** [review]
LGTM - comment about using functional ways of looping rather than for loops but it's WIP so happy to defer.

**Comment by joelstobart-moj** [comment]
@yasinmustafa  I have replaced this with functional alternatives

**Review by yasinmustafa — APPROVED** [review]
lgtm

**Pull request merged**